server/src/com/thoughtworks/go/server/service/ArtifactsService.java

```diff
@@ -126,6 +133,19 @@ public boolean saveOrAppendFile(File dest, InputStream stream) {
         }
     }
 
```

Comment:
Also, I don't see a test covering check for diskSpaceMonitor.isLowOnDisk(), could you add that